src/io/perun.clj

```diff
@@ -27,47 +29,88 @@
       (boot/add-resource tmp)
       boot/commit!))
 
+(defn tmp-by-ext
+  "Returns boot tmpfiles from `fileset` that end with `extensions`.
+  If `extensions` is empty, returns all files."
+  [fileset extensions]
+  (cond->> (boot/ls fileset)
+    (> (count extensions) 0) (boot/by-ext extensions)))
+
+(defn meta-by-ext
+  "Returns perun metadata from `fileset`, filtered by `extensions`.
+  If `extensions` is empty, returns metadata for all files."
+  [fileset extensions]
+  (map (partial pm/meta-from-file fileset) (tmp-by-ext fileset extensions)))
+
+(defn filter-tmp-by-ext
+  "Returns boot tmpfiles from `fileset`. `options` selects files
+  that end with values in the `:extensions` key, filtered by the
+  `:filterer` predicate. If `:extensions` is empty, returns all files."
+  [fileset options]
+  (filter (comp (:filterer options) (partial pm/meta-from-file fileset))
+          (tmp-by-ext fileset (:extensions options))))
+
+(defn filter-meta-by-ext
+  "Returns perun metadata from `fileset`. `options` selects files
+  that end with values in the `:extensions` key, filtered by the
+  `:filterer` predicate. If `:extensions` is empty, returns
+  metadata for all files."
+  [fileset options]
```

Comment:
fair enough.  (you could also use the destructuring to rename the variable without changing the key.)